components/tidb_query/src/codec/mysql/json/binary.rs

```diff
@@ -141,6 +196,23 @@ pub trait JsonEncoder: NumberEncoder {
         Ok(())
     }
 
+    fn write_json_ref_array<'a>(&mut self, data: &[JsonRef<'a>]) -> Result<()> {
```

Comment:
The `write_json_ref_array` has duplicate code with `write_json_array`, I think we should use an unify signature to eliminate them, e.g:
```rust
fn write_json_array<'a>(&mut self, data: impl Iterator<impl AsRef<JsonRef<'a>>>) -> Result<()>
```